### DISCUSSION TITLE
Revert to old project-config behaviour (or an option to anyway)

### DISCUSSION THREAD

**Discussion by sunscreem** [craftcms/cms#8946] [Ideas]
### Description

Here's the scenario:

I add a field to the backend and output that field on the front end. 

I commit the changes.

My colleague pulls those changes locally.

He's a front end developer, so, without logging in to the backend has no way to know there are outstanding project config changes. 

All he sees is a 500 error because the field doesn't exist yet.

Yes - he can run `project-config/apply` or log in to the back end and click the warning, but it's not obvious. Especially in dev mode as there are usually big red-warnings up the top of the dashboard until licences are sorted etc.

### Suggested Fix:

Add a config option something like `requireProjectConfigChangesBeApplied` or something better worded, that reverts back to the previous behaviour or 503'ing the whole site until project.yaml was applied.

**Comment by michtio** [comment]
As a temporary solution, let your colleague create a bash alias ( if he is using CLI ) `alias gp = git pull && ./craft project-config/apply` or even `alias gp = git pull && composer update` and add the composer hook in. That's how we solved it here.

**Comment by sunscreem** [comment]
@michtio Yeah - we added the project-config/apply to the post-merge hook in git as a workaround.

**Comment by sunscreem** [comment]
I wonder if there's any update on this?

**Comment by brandonkelly** [comment]
@sunscreem Not currently, waiting to see if it gains a bit more traction.
